projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -753,6 +754,14 @@ public static function enqueue_block_editor_assets() {
 			);
 		}
 
+		// We need this information in the ai-chat feature, because it cant be activated if instant search is not enabled.
+		if ( Jetpack::is_module_active( 'search' ) ) {
+			$search_plan             = new Search_Plan();
+			$initial_state['search'] = array(
+				'instant_search_enabled' => (bool) $search_plan->supports_instant_search(),
```

Comment:
Good idea!

Reply:
I don't like this global approach. I'm gonna use the `jetpack-search/plan` endpoint

Reply:
What don't you like about it and why do you prefer making an additional request to the site?

Reply:
Because we either:

- Make an additional request for people who DO want to use that block
- Add a parameter to hundreds of millions of sites (everybody) who may or may not want to use that block

Collective waste is much bigger if we add parameter everywhere

Reply:
A new request brings new considerations like: 
- Caching 
- Not calling it from the front-end
- Only calling it in context (when the block is present)
- Error handling 
- State of the block when waiting for response

Reply:
Ok, I don't have strong opinions here :D

Reply:
Done in [8c0bd9e](https://github.com/Automattic/jetpack/pull/33064/commits/8c0bd9ea572c86e3839b07e2a1e6d03a43a8a0b7)